Fixes #3617 Clean busting directory when purging cache

remyperona opened the pull request:
## Description

Add a call to `rocket_clean_cache_busting()` when purging the whole cache, to make sure it's updated correctly.

Fixes #3617 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Is the solution different from the one proposed during the grooming?

No

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] Existing unit tests pass locally with my changes

remyperona committed:
clean busting directory when purging cache

Mai-Saad reviewed (commented):
Currently, clearing cache from the admin bar or from the quick actions in the dashboard tab will clear the dynamic style /js file from busting. but the .gz file won't be removed. However, with the next page visit, the .gz file will be overwritten. @Tabrisrp Do you think this is fine or do we need to do anything here?

remyperona committed:
refactor rocket_clean_cache_busting()

remyperona committed:
add stub for WP_ROCKET_CACHE_BUSTING_PATH

remyperona committed:
add files for cache busting

remyperona committed:
add tests for rocket_clean_cache_busting()

remyperona commented:
@Mai-Saad I updated the code to also clear the .gz files when the function is called.

@wp-media/php Need another code review, as I refactored the code of the function, and also added unit/integration tests for it

Mai-Saad reviewed (approved):
All files are deleted now from busting.
